pkg/util/event_emmiter.go

```diff
@@ -37,7 +39,9 @@ func (ee eventEmitter) EmitEvent(object runtime.Object, eventType, reason, msg s
 		ee.recorder.Event(ee.pod, eventType, reason, msg)
 	}
 
-	if object != nil {
+	// checking object != nil does not work because object is an interface, and nil interface instance is not nil.
+	// We need to check that it's actually nil, using reflection.
+	if t := reflect.ValueOf(object); t.Kind() != reflect.Ptr || !t.IsNil() {
```

Comment:
not that it make a difference, but maybe use the exact same code as in main?
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2011/files#diff-04c60da2651653a56d9a3e11edaa1694261a63200172901315131ea465758747R44
(i.e. `Pointer` vs. `Ptr`)

Reply:
Couldn't do it before. Pointer added at go 1.18. This branch was compiled with 1.17. I just fixed it, but it wasn't ready on time.